File: EvolutionaryGenerator_v2.1_Python/metah/selector.py
"""Translation of ``mx.tec.meta.Selector``."""

from abc import ABC, abstractmethod

from javacompat import Random
from metah.individual import Individual


class Selector(ABC):
    """Defines the methods to create and handle selection operators to be used by a genetic algorithm.

    :author: Paola Azeneth Castillo Gutiérrez (Python translation)
    :original author: Jose Carlos Ortiz Bayliss (jcobayliss@tec.mx)
    :original version: 2.0
    """

    def __init__(self, seed: int):
        """Creates a new instance of ``SelectionOperator``.

        :param seed: The seed for the random number generator to be used by this selector.
        """
        #: The random number generator to be used in all random operations within this selector.
        self.random = Random(seed)

    @abstractmethod
    def select(self, population: list, objective) -> list:
        """Selects the solutions to be used for crossover.

        :param population: The solutions contained in the current population.
        :param objective: The objective of the evolutionary process regarding the
            objective function (maximize or minimize).
        :return: The solutions to be used for crossover.
        """

File: EvolutionaryGenerator_v2.1_Java/MetaH/src/mx/tec/meta/Selector.java
package mx.tec.meta;

import mx.tec.meta.Individual;
import java.util.List;
import java.util.Random;
import mx.tec.meta.ga.GeneticAlgorithm.Objective;

/**
 * Defines the methods to create and handle selection operators to be used by a genetic algorithm.
 * 
 * @author Jose Carlos Ortiz Bayliss (jcobayliss@tec.mx)
 * @version 2.0
 */
public abstract class Selector {
       
    /**
     * The random number generator to be used in all random operations within this selector.
     */
    protected final Random random;
    
    /**
     * Creates a new instance of <code>SelectionOperator</code>.
     * 
     * @param seed The seed for the random number generator to be used by this selector.
     */
    protected Selector(long seed) {        
        random = new Random(seed);
    }   

    /**
     * Selects the solutions to be used for crossover.
     * 
     * @param population The solutions contained in the current population.
     * @param objective The objective of the evolutionary process regarding the objective function (maximize or minimize).
     * @return The solutions to be used for crossover.
     */
    public abstract Individual[] select(List<Individual> population, Objective objective);
}
